CodeGenerator/Sources/CodeGenerator/main.swift

```diff
@@ -4,9 +4,25 @@ import Dispatch
 import Stencil
 import PathKit
 
+extension String {
+    func writeIfChanged(toFile: String, atomically: Bool, encoding: String.Encoding) throws -> Bool {
+        do {
+            let original = try String(contentsOfFile: toFile)
+            guard original != self else { return false }
```

Comment:
We read the complete file before we write it? I'd assume that this should be far more expensive than just rewriting the file.

Reply:
The writeIfChanged() has nothing to do with the performance of the CodeGenerator. It is to do with the effect it has on the XCode build. If I write a file, regardless of whether it changes, the build needs to recompile it. Also XCode does stuff when you change files from underneath it which can have a major performance hit of your Mac. 

As an aside the load to check it it has changed has a minor hit on performance. 

I’ll add a comment detailing this.

Reply:
The writeIfChanged() has nothing to do with the performance of the CodeGenerator. It is to do with the effect it has on the XCode build. If I write a file, regardless of whether it changes, the build needs to recompile it. Also XCode does stuff when you change files from underneath it which can have a major performance hit of your Mac. 

As an aside the load to check it it has changed has a minor hit on performance. 

I’ll add a comment detailing this.